src/include/OpenImageIO/sysutil.h

```diff
@@ -116,6 +120,19 @@ physical_concurrency();
 OIIO_API size_t
 max_open_files();
 
+/// Return a string containing a readable stack trace from the point where
+/// it was called. Return an empty string if not supported on this platform.
+OIIO_API std::string
```

Comment:
I'm not totally clear how the user can gain access to this string.

If the program crashes, the default handler is invoked and the program exits immediately afterwards, no?

Reply:
You don't have to wait for a crashing fault to ask for a stack trace!

Reply:
Oh I see, you are thinking of regular debugging cases too.

Reply:
The main reason I did this is to be able to increase the set of circumstances in which users can give me clues about where something crashed, even if they are not debugger-savvy or I can't reproduce the crash on my end.

For the command line tools, installing a signal handler is enough.

But it was so little extra work to full expose it as a string, I thought it would be helpful to open up the possibilities of a new kind of rich diagnostic error for certain nonfatal error situations (but that are unexpected enough that we'd really want to know more about the circumstances that led to them). You could build an interesting kind of non-aborting assertion out of it, for example, that tells you much more about exactly what the call sequence was, without having to catch it live in a debugger at the moment of the abort().